### ISSUE TITLE
About the position of the distillation token

### ISSUE THREAD

**Issue by sihunbaek** [facebookresearch/deit#79]
When I read the paper, I assume that the distillation token's position is at the last in the line, as described in the figure. But in the code, distillation token is right behind the class token. Did I misunderstand the paper? Or is there any reason?

-------------------------
        cls_tokens = self.cls_token.expand(B, -1, -1)  # stole cls_tokens impl from Phil Wang, thanks
        dist_token = self.dist_token.expand(B, -1, -1)
        x = torch.cat((cls_tokens, dist_token, x), dim=1)
in modely.py

**Comment by TouvronHugo** [comment]
Hi @sihunbaek ,
No, for the sake of clarity we have represented it like this on the figure because the position of the tokens does not have any importance. The position information is learned and injected by the position encoding.
Best,
Hugo
